src/Api/Helper.php

```diff
@@ -28,6 +31,85 @@ public function __construct(Api $api)
     }
 
     /**
+     * Upload file
+     *
+     * Returns file key to use for list import, email content, etc.
+     *
+     * @param string $path
+     * @return string file key
+     */
+    public function uploadFile($path)
+    {
+        if (!is_readable($path)) {
+            throw new \InvalidArgumentException('File path is not readable: ' . $path);
+        }
+        $basename = basename($path);
+        $mime = (new \finfo(FILEINFO_MIME_TYPE))->file($path);
+        if ($mime === false) {
+            throw new \RuntimeException("MIME type could not be determined");
+        }
+
+        // Initialise
+        /** @var string $fileKey */
+        $fileKey = $this->api->file_upload->initialise()->key;
+
+        // Build request
+        $this->setConnectionConfig($this->api->getConfig());
+        $headers = $this->getHeaders();
+
+        $handleurl = ($this->useSsl ? 'https' : 'http') .
+            '://' .
+            $this->host .
+            '/api/json/file_upload';
+
+        $params = array(
+            'method' => 'handle',
+            'key'    => $fileKey,
+            'file'   => "@{$path};filename={$basename};type={$mime}"
```

Comment:
Yeah, but for BC for php54 we need to keep using @file.

In php55 I get a E_DEPRECATED but it functions without issue.